core/src/main/java/io/kestra/core/contexts/KestraBeansFactory.java

```diff
@@ -2,16 +2,66 @@
 
 import io.kestra.core.plugins.DefaultPluginRegistry;
 import io.kestra.core.plugins.PluginRegistry;
+import io.kestra.core.storages.StorageInterface;
+import io.kestra.core.storages.StorageInterfaceFactory;
+import io.micronaut.context.annotation.Bean;
+import io.micronaut.context.annotation.ConfigurationProperties;
 import io.micronaut.context.annotation.Factory;
 import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.annotation.Value;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.convert.format.MapFormat;
+import io.micronaut.core.naming.conventions.StringConvention;
+import jakarta.inject.Inject;
 import jakarta.inject.Singleton;
+import jakarta.validation.Validator;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
 
 @Factory
 public class KestraBeansFactory {
 
+    @Inject
+    Validator validator;
+
+    @Inject
+    StorageConfig storageConfig;
+
+    @Value("${kestra.storage.type}")
+    Optional<String> storageType;
+
     @Requires(missingBeans = PluginRegistry.class)
     @Singleton
     public PluginRegistry pluginRegistry() {
         return DefaultPluginRegistry.getOrCreate();
     }
+
+    @Requires(missingBeans = StorageInterface.class)
+    @Requires(property = "kestra.storage.type")
+    @Singleton
+    @Bean(preDestroy = "close")
+    public StorageInterface storageInterface(final PluginRegistry pluginRegistry) throws IOException {
+        String pluginId = storageType.get();
+        return StorageInterfaceFactory.make(pluginRegistry, pluginId, storageConfig.getStorageConfig(pluginId), validator);
+    }
```

Comment:
I would remove the `@Requires` annotation and use `storageType.orElseThrow` with a message explaining that no storage is configured. We sometime have people asking question about Kestra refuses to start that ends up being an issue with the storage configuration so a good error message may help those people to understand what's going on.